clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10717,6 +10717,10 @@ getTripleBasedSYCLPostLinkOpts(const ToolChain &TC, const JobAction &JA,
       (IsAOT || NewOffloadDriver))
     addArgs(PostLinkArgs, TCArgs,
             {"-generate-device-image-default-spec-consts"});
+  bool IsUsingLTO = TC.getDriver().isUsingLTO(/*IsDeviceOffloadAction=*/true);
+  auto LTOMode = TC.getDriver().getLTOMode(/*IsDeviceOffloadAction=*/true);
+  if (IsUsingLTO && LTOMode == LTOK_Thin)
+    addArgs(PostLinkArgs, TCArgs, {"-embed-aux-info-as-metadata"});
```

Comment:
This doesn't look like it needs any Triple information to resolve - can you move into `getNonTripleBasedSYCLPostLinkOpts`?

Reply:
oh yeah sorry, fixed in latest commit, thanks.